scripts/prepDataForGephi.py

```diff
@@ -0,0 +1,330 @@
+
+# If the user didn't want to filter their input data, then we just use the original directory
+if(filteredDirectoryPath == ""):
+    filteredDirectoryPath = directoryPath
+
+# Create the SNP mapping from our files
+for filename in os.listdir(filteredDirectoryPath):
+    # Make sure to ignore any DS_Store files that are automatically generated by filtration
+    if(filename.split(".")[1] != "DS_Store"):
+        f = open(filteredDirectoryPath + filename, 'r')
+        lines = f.readlines()
+        indexforPhenotype = None
+        indexforSnpName = None
+        indexforPValue = None
+
+        snpsForThisPhecodeList = []
+        snpsForThisPhecodeSet = set()
+
+        # Figure out where phenotype name, snp name, and p-value are in the filtered data that are being processed
+        firstLine = lines[0]
+        firstLineAsArray = firstLine.split(' ')
+        for i in range(0, len(firstLineAsArray)):
+            if(firstLineAsArray[i] == phecodeColName):
+                indexforPhenotype = i
+            elif(firstLineAsArray[i] == snpColName):
+                indexforSnpName = i
+            elif(firstLineAsArray[i] == pValueColName):
+                indexforPValue = i
+
+        for elem in lines:
+            # Get the name of the phenotype
+            phecode = elem.split(fileDelimiter)[indexforPhenotype]
+            # Get the name and p-value of the SNP
+            snpName = elem.split(fileDelimiter)[indexforSnpName]
+            # If this isn't the first row of the file, add this tuple to our list/set
+            if(snpName != snpColName):
+                snpsForThisPhecodeSet.add(snpName)
+
+            if(indexforPValue != None):
+                snpPValue = elem.split(fileDelimiter)[indexforPValue]
+
+                # If this isn't the first row of the file, add this tuple to our list/set
+                if(snpName != snpColName):
+                    snpPValueFloat = float(snpPValue)
+                    snpsForThisPhecodeList.append((snpName, snpPValueFloat))
+        
+        # If SNPs have been found for this phecode, we add it to our mappings
+        if(len(snpsForThisPhecodeSet) > 0):
+            # Set is used for the edge map, to find overlaps of associations between phenotypes;
+            snpMappingsSet[phecode] = snpsForThisPhecodeSet
+
+        if(len(snpsForThisPhecodeList) > 0):
+        # List is used for the node map to present a sorted list of nodes
+            # Sort the list by p-value (i.e., by the second element of each tuple)
+            snpsForThisPhecodeList.sort(key = lambda x: float(x[1]), reverse = True)
+            snpsForThisPhecodeList.reverse()
+            # Get a list of the now sorted SNP IDs
+            snpNamesSorted = [i[0] for i in snpsForThisPhecodeList]
+            snpMappingsList[phecode] = snpNamesSorted
+
+print("Finished identifying phenotype to SNP mappings.")
+
+
+
+# If we want to create a node map file
+if(nodeMapInputPath != "" and nodeMapOutputPath != ""):
```

Comment:
the node map creation should work even if the user does not give any input. Can we get rid of the input completely?

Reply:
The idea here is that "node map creation" takes an input node map and just appends another column that includes associated SNPs for each row. So, it does need an input node map. Let's discuss.